pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -179,10 +178,12 @@ func (h *handler) deployK3sBasedUpgradeController(clusterName, updateVersion str
 			}
 		}
 
-		desiredApp := app.DeepCopy()
-		if desiredApp.Spec.Answers == nil {
```

Comment:
@chiukapoor Why are we removing Answers field check here? I first thought it's cuz it won't be nil as we initialized map in app spec but looks like we removed it there too.

Reply:
Context of this removal is lost after rebasing with the latest changes pushed by 1.29 PR where some extent of PSP removal was done. This [commit](https://github.com/rancher/rancher/commit/bff2d6827892a38c00ec1b2fd648aae9eeb759c9#diff-e2f8bb33302c1afb0e07d998784b234c950de2adf0eb0a634784c5cefae62e8c) from the 1.29 PR is the other part of this PR.

The `Answers` field was only being used to cater the `PSPAnswersField` in it. Now as the `PSPAnswersField` is removed we may safely remove the `Answers` field as well.

We are kind of reverting https://github.com/rancher/rancher/pull/40445 PR which add PSP checks and `Answers` field.